scripts/get-workspace-map.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+# get-workspace-map.sh context-1 context-2... context-n
+#   Read Namespace resources and generate a JSON object mapping from namespaces
+#   to workspace based on the value of label toolchain.dev.openshift.com/space.
+#   This script assumes `oc` is preconfigured with all the required clusters
+#   and could be found in $PATH
+#
+set -o pipefail -o errexit -o nounset
+
+out={}
+for context in "$@"
+do
+  oc config use-context "$context"
+  new=$(
+    oc get namespaces \
+      --selector=toolchain.dev.openshift.com/type=tenant \
+      -o=go-template \
+      --template=$'{
+        {{- $comma := false}}
+        {{- range .items}}
+            {{- $namespace := .metadata.name }}
+            {{- $workspace := index .metadata.labels "toolchain.dev.openshift.com/space" }}
+            {{- if $comma}},{{end}}
+            {{- $comma = true -}}
+            "{{$namespace}}":"{{ $workspace }}"
+        {{- end -}}}\n'
+  )
+  out=$(echo "$out" "$new"  | jq -s '.[0] * .[1]')
```

Comment:
I think there is a way to stream all the different cluster maps into jq at one and have it merge them all instead of running a different jq command for each pair (IIRC it has at least a `reduce` function)

This will also avoid relying on shell buffers which are prone to be limited.

Reply:
Looking at the jq docs: while we have reduce, the add function may be sufficient in this case